modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java

```diff
@@ -314,4 +315,76 @@ private String getAuthzHeader() {
     }
 
+    private JSONObject getPayload(HttpResponse response) throws IOException {
+
+        BufferedReader rd = new BufferedReader(new InputStreamReader(response.getEntity().getContent()));
+        Object responseObj = JSONValue.parse(rd);
+        EntityUtils.consume(response.getEntity());
+        return (JSONObject) responseObj;
+    }
+
+    @Test(alwaysRun = true, groups = "wso2.is", priority = 9, description = "Create a service provider with " +
+            "additional OIDC properties")
+    public void testCreateServiceProviderRequestWithAdditionalParameters() throws IOException {
+
+        HttpPost request = new HttpPost(getPath());
+        request.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
+        request.addHeader(HttpHeaders.CONTENT_TYPE, OAuthDCRMConstants.CONTENT_TYPE);
+
+        JSONArray grantTypes = new JSONArray();
+        grantTypes.add(OAuthDCRMConstants.GRANT_TYPE_AUTHORIZATION_CODE);
+        grantTypes.add(OAuthDCRMConstants.GRANT_TYPE_IMPLICIT);
+
+        JSONArray redirectURI = new JSONArray();
+        redirectURI.add(OAuthDCRMConstants.REDIRECT_URI);
+
+        JSONObject obj = new JSONObject();
+        obj.put(OAuthDCRMConstants.CLIENT_NAME, "DCR_1");
+        obj.put(OAuthDCRMConstants.GRANT_TYPES, grantTypes);
+        obj.put(OAuthDCRMConstants.REDIRECT_URIS, redirectURI);
+        obj.put(OAuthDCRMConstants.TOKEN_AUTH_METHOD, "private_key_jwt");
+        obj.put(OAuthDCRMConstants.TOKEN_AUTH_SIGNATURE_ALGORITHM, "PS256");
+        obj.put(OAuthDCRMConstants.SECTOR_IDENTIFIER_URI, "https://mocki.io/v1/04b49547-0ae2-4049-8d1c-42648e633001");
+        obj.put(OAuthDCRMConstants.ID_TOKEN_SIGNATURE_ALGORITHM, "PS256");
+        obj.put(OAuthDCRMConstants.ID_TOKEN_ENCRYPTION_ALGORITHM, "RSA-OAEP");
+        obj.put(OAuthDCRMConstants.ID_TOKEN_ENCRYPTION_METHOD, "A128GCM");
+        obj.put(OAuthDCRMConstants.REQUEST_OBJECT_SIGNATURE_ALGORITHM, "PS256");
+        obj.put(OAuthDCRMConstants.REQUEST_OBJECT_ENCRYPTION_ALGORITHM, "RSA-OAEP");
+        obj.put(OAuthDCRMConstants.REQUEST_OBJECT_ENCRYPTION_METHOD, "A128GCM");
+        obj.put(OAuthDCRMConstants.TLS_SUBJECT_DN, "dfrrfc");
+        obj.put(OAuthDCRMConstants.IS_SIGNED_REQUEST_OBJECT, true);
+        obj.put(OAuthDCRMConstants.IS_PUSH_AUTH, true);
+        obj.put(OAuthDCRMConstants.IS_CERTIFICATE_BOUND_ACCESS_TOKEN, true);
+        obj.put(OAuthDCRMConstants.SUBJECT_TYPE, "pairwise");
+        obj.put(OAuthDCRMConstants.JWKS_URI, "https://localhost:9443/oauth2/jwks");
+
+        StringEntity entity = new StringEntity(obj.toJSONString());
+        request.setEntity(entity);
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 201, "Service Provider " +
+                "has not been created successfully");
+
+        JSONObject createResponsePayload  = getPayload(response);
```

Comment:
Lets assert that all the attributes that should be returned has returned as well.

Reply:
fixed